ui/components/app/assets/token-list/token-list.tsx

```diff
@@ -1,109 +1,233 @@
+          const { isNative, address, decimals } = token;
+          const balance =
+            calculateTokenBalance({
+              isNative,
+              chainId,
+              address,
+              decimals,
+              nativeBalances,
+              selectedAccountTokenBalancesAcrossChains,
+            }) || '0';
+
+          const tokenFiatAmount = calculateTokenFiatAmount({
+            token,
+            chainId,
+            balance,
+            marketData,
+            currencyRates,
+          });
+
+          // Append processed token with balance and fiat amount
+          tokensWithBalance.push({
+            ...token,
+            balance,
+            tokenFiatAmount,
+            chainId,
+            string: String(balance),
+          });
+        });
+      },
+    );
+
+    return tokensWithBalance;
   };
 
-  const sortedTokens = useMemo(() => {
-    // TODO filter assets by networkTokenFilter before sorting
-    return sortAssets(
-      [nativeTokenWithBalance, ...tokensWithBalances],
-      tokenSortConfig,
+  const sortedFilteredTokens = useMemo(() => {
+    const consolidatedTokensWithBalances = consolidatedBalances();
+    const filteredAssets = filterAssets(consolidatedTokensWithBalances, [
+      {
+        key: 'chainId',
+        opts: tokenNetworkFilter,
+        filterCallback: 'inclusive',
+      },
+    ]);
+
+    const { nativeTokens, nonNativeTokens } = filteredAssets.reduce<{
+      nativeTokens: TokenWithFiatAmount[];
+      nonNativeTokens: TokenWithFiatAmount[];
+    }>(
+      (acc, token) => {
+        if (token.isNative) {
+          acc.nativeTokens.push(token);
+        } else {
+          acc.nonNativeTokens.push(token);
+        }
+        return acc;
+      },
+      { nativeTokens: [], nonNativeTokens: [] },
     );
+    const assets = [...nativeTokens, ...nonNativeTokens];
+    return sortAssets(assets, tokenSortConfig);
   }, [
-    tokensWithBalances,
     tokenSortConfig,
     tokenNetworkFilter,
     conversionRate,
     contractExchangeRates,
+    currentNetwork,
+    selectedAccount,
+    selectedAccountTokensChains,
+    newTokensImported,
   ]);
 
   useEffect(() => {
-    if (!loading) {
+    if (sortedFilteredTokens) {
       endTrace({ name: TraceName.AccountOverviewAssetListTab });
     }
-  }, [loading]);
-
-  return loading ? (
-    <Box
-      display={Display.Flex}
-      alignItems={AlignItems.center}
-      justifyContent={JustifyContent.center}
-      padding={7}
-      data-testid="token-list-loading-message"
-    >
-      {t('loadingTokens')}
-    </Box>
-  ) : (
+  }, [sortedFilteredTokens]);
+
+  // Displays nativeToken if provided
+  if (nativeToken) {
+    return React.cloneElement(nativeToken as React.ReactElement);
+  }
+
+  // TODO: We can remove this string. However it will result in a huge file 50+ file diff
+  // Lets remove it in a separate PR
+  if (sortedFilteredTokens === undefined) {
+    console.log(t('loadingTokens'));
+  }
```

Comment:
Sorry for this. But, I didn't want to blow this PR up another 50+ files for updating an unused locale string. I will remove this in a quick follow PR.